samples-generator/bin/3d-tiles-samples-generator.js

```diff
@@ -919,6 +942,17 @@ function saveBatchedTileset(tilesetName, tileOptions, tilesetOptions) {
             var b3dm = result.b3dm;
             var batchTableJson = result.batchTableJson;
             tilesetOptions.properties = getProperties(batchTableJson);
+
+            if (tilesetOptions.contentDataUri) {
+                return saveTile(tilePath, b3dm, gzip).then(function() {
+                    return dataUri(tilePath).then(function (content) {
+                        // TODO delete b3dm file
+                        tilesetOptions.tileName = content;
+                        return saveTilesetJson(tilesetPath, createTilesetJsonSingle(tilesetOptions), prettyJson);
+                    });
```

Comment:
This section is a little awkward. Why not just create a data uri (synchronously) from `b3dm` rather than saving the file first?

Reply:
Also consider renaming `tilesetOptions.tileName` to `tilesetOptions.contentUri` throughout, and update documentation in `createTilesetJsonSingle`.